File: mex/editor/client/scripts/import-theme.mjs

// write script so it transforms the variable names to match actual usage? currently missmatched
import { readFileSync, writeFileSync, mkdirSync, existsSync } from 'node:fs';
import { dirname } from 'node:path';

const INPUT = process.argv[2] ?? 'theme-export/light.css';
const OUTPUT = process.argv[3] ?? 'src/theme-tokens.css';

if (!existsSync(INPUT)) {
  console.error(`Input file not found: ${INPUT}`);
  console.error(`Usage: node scripts/import-theme.mjs <input.css> <output.css>`);
  process.exit(1);
}

const input = readFileSync(INPUT, 'utf8');

const output = input
  .replace(/--md-sys-color-([a-z0-9-]+):/g, '--mat-sys-$1:')
  .replace(/^[^{]+\{/, ':root {'); // replace whatever the top selector is, not just ".light"

// script won't fail if src/styles/ doesn't exist yet
mkdirSync(dirname(OUTPUT), { recursive: true });
writeFileSync(OUTPUT, output);

console.log(`Wrote ${OUTPUT} from ${INPUT}`);

File: mex/editor/client/scripts/import-theme.py
#!/usr/bin/env python3

import logging
import re
import sys
from pathlib import Path

# Named argument indexes to avoid magic numbers (PLR2004)
INPUT_ARG_INDEX = 1
OUTPUT_ARG_INDEX = 2


def main(argv: list[str]) -> int:
    """Convert a theme CSS file to the format used by the editor."""
    logger = logging.getLogger(__name__)
    input_path = (
        Path(argv[INPUT_ARG_INDEX])
        if len(argv) > INPUT_ARG_INDEX
        else Path("theme-export/light.css")
    )
    output_path = (
        Path(argv[OUTPUT_ARG_INDEX])
        if len(argv) > OUTPUT_ARG_INDEX
        else Path("src/theme-tokens.css")
    )

    if not input_path.exists():
        logger.error("Input file not found: %s", input_path)
        logger.error("Usage: python scripts/import-theme.py <input.css> <output.css>")
        return 1

    text = input_path.read_text(encoding="utf-8")

    # Replace variable prefix
    text = re.sub(r"--md-sys-color-([a-z0-9-]+):", r"--mat-sys-\1:", text)

    # Replace the top selector (whatever it is) with :root {
    text = re.sub(r"^[^{]+\{", ":root {", text, count=1, flags=re.DOTALL)

    output_path.parent.mkdir(parents=True, exist_ok=True)
    output_path.write_text(text, encoding="utf-8")

    logger.info("Wrote %s from %s", output_path, input_path)
    return 0


if __name__ == "__main__":
    raise SystemExit(main(sys.argv))
